spaceship/docs/Authentication.md

```diff
@@ -0,0 +1,29 @@
+# Authentication with spaceship
+
+As spaceship talks to Apple's APIs, most requests need to be authenticated with your Apple ID that you use to log in to https://developer.apple.com/. You start the login with a simple call to `Spaceship.login`.
+
+## Credentials
+
+The `Spaceship.login` method accepts `username` and `password` as parameters or will ask for them interactively if not provided. It will also try to retrieve them via [`CredentialsManager`](https://github.com/fastlane/fastlane/tree/master/credentials_manager), which supports environment variables `FASTLANE_USER` and `FASTLANE_PASSWORD` .
+
+Login with API tokens is not supported yet as API access is not generally available. Spaceship is also still using only older APIs that do not need API tokens for authentication yet.
+
+## Two-factor Authentication
+
+If your Apple ID has Two-factor Authentication enabled spaceship will also ask you for the security code that was pushed to your devices after succesfully entering username and password.
+
+If you can not access any of your trusted devices, or just prefer it, you can also switch to the SMS based flow by entering `sms` at this prompt. spaceship will then present you with a selection of your trusted phone numbers, and after choosing you can enter the security code you were sent by SMS.
+
+Spaceship also supports legacy [Two-step verification](https://support.apple.com/en-us/HT204152) that is still active for some Apple IDs.
+
+### Avoid 2FA via additional account
+
+If you want to avoid dealing with the required interaction of using a Two-factor Authentication enabled account (for example because you want to use spaceship on an automated or CI system that does not support interaction), you have to create an additional account with the required rights.
```

Comment:
```suggestion
If you want to avoid dealing with the required interaction of using a Two-factor Authentication enabled account (for example because you want to use _spaceship_ on an automated or CI system that does not support interaction), you have to create an additional account with the required rights.
```